### PR TITLE
AQ-276 Reusable Search Bar Component

### PR DESCRIPTION
**Pull request by jasonmlam**
The purpose of this PR is to create a search component that can be used on all pages of the web application and can also be used for selecting a stock ticker in the Create Investor Page for the future.

Outline of changes:
- Dropdown for stock options through click or "Enter" keypress
- Ability for user to traverse through search options through arrow up/down keys
- Selection by clicking on search options 
- Brings user to a new page displaying selected search option

### CODE REVIEW

**Review by ryPattillo — APPROVED** [review]

File: src/components/pages/HomePage.js [hunk 1]
```diff
@@ -1,21 +1,22 @@
 import { React } from "react";
 import Navbar from "../reusable/NavBar";
 import Sidebar from "../reusable/SideBar";
-import { useAuthenticator } from "@aws-amplify/ui-react";
+// import { useAuthenticator } from "@aws-amplify/ui-react";
 
 const HomePage = () => {
-  const { user } = useAuthenticator((context) => [context.user]);
+  // const { user } = useAuthenticator((context) => [context.user]);
```
Comment (timanese) [inline src/components/pages/HomePage.js:7]: are these needed anymore ?

File: src/components/pages/HomePage.js [hunk 2]
```diff
@@ -1,21 +1,22 @@
 import { React } from "react";
 import Navbar from "../reusable/NavBar";
 import Sidebar from "../reusable/SideBar";
-import { useAuthenticator } from "@aws-amplify/ui-react";
+// import { useAuthenticator } from "@aws-amplify/ui-react";
```
Comment (timanese) [inline src/components/pages/HomePage.js:4]: needed?

Comment (ryPattillo) [reply src/components/pages/HomePage.js]: Resolved

File: src/components/reusable/SearchBar.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+          {searchResults.map((result, index) => (
+            <p
+              className={`px-4 py-2 hover:bg-another-gray cursor-pointer ${
+                highlightedIndex === index
+                  ? "bg-another-gray border-l border-light-gray"
+                  : ""
+              }`}
+              key={result}
+              onClick={() => handleClick(result)}
+              onKeyDown={() => handleKey(result)}
```
Comment (timanese) [inline src/components/reusable/SearchBar.js:120]: is this needed? currently this function is already being done in the input box, those <p/Ps are not the ones that are getting new values for search.

File: src/components/reusable/SearchBar.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  // The current search value entered
+  const [searchValue, setSearchValue] = useState("");
+
+  // Which item in the drop down the user has selected
+  const [highlightedIndex, setHighlightedIndex] = useState(-1);
+
+  // The search results to be displayed in the dropdown
+  const [searchResults, setSearchResults] = useState([]);
+
+  const searchRef = useRef(null);
```
Comment (timanese) [inline src/components/reusable/SearchBar.js:22]: @jasonmlam what is this used for?

**Review by timanese — APPROVED** [review]

**Pull request merged**